tests/TestCase/ORM/QueryTest.php

```diff
@@ -3412,6 +3412,105 @@ public function testLeftJoinWithAutoFields()
         $this->assertCount(3, $results);
     }
 
+    /**
+     * Test leftJoinWith and contain on optional association
+     *
+     * @return void
+     */
+    public function testLeftJoinWithAndContainOnOptionalAssociation()
+    {
+        $table = $this->getTableLocator()->get('Articles', ['table' => 'articles']);
+        $table->belongsTo('Authors');
+        $newArticle = $table->newEntity([
+            'title' => 'Fourth Article',
+            'body' => 'Fourth Article Body',
+            'published' => 'N',
+        ]);
+        $table->save($newArticle);
+        $results = $table
+            ->find()
+            ->disableHydration()
+            ->contain('Authors')
+            ->leftJoinWith('Authors')
+            ->all();
+        $expected = [
+            [
+                'id' => 1,
+                'author_id' => 1,
+                'title' => 'First Article',
+                'body' => 'First Article Body',
+                'published' => 'Y',
+                'author' => [
+                    'id' => 1,
+                    'name' => 'mariano',
+                ],
+            ],
+            [
+                'id' => 2,
+                'author_id' => 3,
+                'title' => 'Second Article',
+                'body' => 'Second Article Body',
+                'published' => 'Y',
+                'author' => [
+                    'id' => 3,
+                    'name' => 'larry',
+                ],
+            ],
+            [
+                'id' => 3,
+                'author_id' => 1,
+                'title' => 'Third Article',
+                'body' => 'Third Article Body',
+                'published' => 'Y',
+                'author' => [
+                    'id' => 1,
+                    'name' => 'mariano',
+                ],
+            ],
+            [
+                'id' => 4,
+                'author_id' => null,
+                'title' => 'Fourth Article',
+                'body' => 'Fourth Article Body',
+                'published' => 'N',
+                'author' => null,
+            ],
+        ];
+        $this->assertEquals($expected, $results->toList());
+        $table->deleteAll([]);
+        $newArticle = $table->newEntity([
+            'title' => 'Fourth Article',
+            'body' => 'Fourth Article Body',
+            'published' => 'N',
+        ]);
+        $table->save($newArticle);
+        $results = $table
+            ->find()
+            ->disableHydration()
+            ->contain('Authors')
+            ->leftJoinWith('Authors')
+            ->all();
+        $expected = [
+            [
+                'id' => 5,
+                'author_id' => null,
+                'title' => 'Fourth Article',
+                'body' => 'Fourth Article Body',
+                'published' => 'N',
+                'author' => null,
+            ],
+        ];
```

Comment:
What is this test covering that the first insert/read isn't? They look very similar to me.

Reply:
The first test covers when there's result with and without the association, some associations are null, and the second covers when there's no results with association, all associations are null.

Reply:
I made some changes to tests to remove the second insert doing a query only on null associations.